### PR TITLE
Global state query

### PR DESCRIPTION
**Pull request by phlptp**
### Summary

If merged this pull request will add a `global_state` query and make some updates and fixes to query mechanics during shutdown.

### CODE REVIEW

**Comment by phlptp** [comment]
- [ ] documentation

**Comment by phlptp** [comment]
Fix #1604

**Comment by phlptp** [comment]
Addresses #1605  as well

**Review by trevorhardy — COMMENTED** [review]
Minor quibble with terminology; I left a comment and will leave it to you to decide if there's anything that we can or should do about it.

**Review by nightlark — APPROVED** [review]

File: src/helics/core/CommonCore.cpp [hunk 1]
```diff
@@ -57,6 +57,36 @@ const std::string& state_string(operation_state state)
...
+    static const std::string dis{"disconnected"};
+    static const std::string exec{"executing"};
+    static const std::string term{"terminating"};
+    static const std::string unk{"unknown"};
+
+    switch (state) {
+        case federate_state::HELICS_CREATED:
+            return c1;
+
+        case federate_state::HELICS_INITIALIZING:
```
[the remark this answers is not in the document]
Comment (phlptp) [reply src/helics/core/CommonCore.cpp:74]: This is my mistake, I am fixing it.  good catch.

**Review by trevorhardy — APPROVED** [review]

**Pull request merged**